js/main.js

```diff
@@ -0,0 +1,504 @@
+let maxBoardSize = 520;													//максимальний розмір дошки у px, висота або ширина
+let tileSize;															//розмір плитки у пікселях
+let tileMoveTime = 90;													//час на пересування плитки
+let totalRowsField = document.querySelector("#totalRowsField");
+let totalColumnsField = document.querySelector("#totalColumnsField");
+let totalRows;
+let totalColumns;
+let tilesQuantity;														//загальна кількість плиток
+let emptyTileindex;														//індекс порожньої клітини
+let stopwatchHandler;
+let stopwatchCounter;
+let movesRecordsData;													//таблиця рекордів по кількості ходів
+let timeRecordsData;													//таблиця рекордів по часу
+let currentBoardDataKey;												//ключ розміру ігрового поля за яким зберігаються рекорди
+let localStorageMovesRecordsKey = "game15movesRecords";					//ключ за яким зберігаються рекорди по кількості ходів
+let localStorageTimeRecordsKey = "game15timeRecords";					//ключ за яким зберігаються рекорди часу
+let savedPositionData;													//дані позиції для зберігання
+let localStoragePositionKey = "game15CurrentPosition";					//ключ за яким зберігається позиція плиток
+let recordsRowTemplate = document.querySelector("#recordsRowTemplate");
+let gamePausedFlag = false;
+
+let board = document.querySelector("#board");
+let pausedElement = document.querySelector("#pausedElement");
+let newGameBtn = document.querySelector("#newGameBtn");
+let pauseBtn = document.querySelector("#pauseBtn");
+let movesCountField = document.querySelector("#movesCountField");
+let timeCountField = document.querySelector("#timeCountField");
+let movesRecordsField = document.querySelector("#movesRecords");
+let timeRecordsField = document.querySelector("#timeRecords");
+
+let tiles;												//масив всіх плиток
+let puzzleIsSolved;											//перевірка чи розв'язана головоломка
+
+addMoveTilesEvents();
+restoreCurrentTilesPosition();
+makeStartBoardAndOptions(savedPositionData);
+updateRecordsDataFromStorage();
+updateRecordsFields();
+
+totalRowsField.addEventListener("change", function(){
```

Comment:
Move all of listeners to separate method this you call only once during opening your app. Like render method